### PR TITLE
enh(stream connector): Improve streamconnector onboarding (Monic-201 and monic-202)

### PR DESCRIPTION
**Pull request by omercier**
## Description

Changes were made on two aspects:

### In Broker BBDO documentation

* some tables were in HTML, others in improper markdown format, it was painful to edit them, so I tidied it up a little
* host and service status events are the most used in LUA stream connector develeopment but their fields lacked a description, so I completed it as much as I could.

### In the stream connector specific doc & tutorial

* changed the plan to something more didactic
* tried to propose more relevant scripts for each step in the progression
* use broker cache API v2
* removed some lua language related hints that I found were too numerous

I'm not fully happy with this but I hope it actually improves the documentation.

## Target serie

- [x] 20.04.x
- [x] 20.10.x (master)

### CODE REVIEW

**Comment by cgagnaire** [comment]
Please rebase your branch on master.

File: en/developer/developer-broker-stream-connector.md [hunk 1]
```diff
@@ -293,62 +293,66 @@ get hostnames, etc...
...
+4.  `get_hostgroup_name(id)` that gets from the cache the host group name of
     the given id. This function returns a string or *nil* otherwise.
-5.  ``get_hostgroups(host_id)`` that gets the list of host groups containing the
+5.  `get_hostgroups(host_id)` that gets the list of host groups containing the
     host corresponding to *host_id*. The return value is an array of objects,
     each one containing two fields, *group_id* and *group_name*.
-6.  ``get_hostname(id)`` that gets from the cache the host name corresponding to
+6.  `get_hostname(id)` (**API v1 only**) that gets from the cache the host name corresponding to
     the given host id. This function returns a string with the host name or
     *nil* otherwise.
```
Comment (bouda1) [inline en/developer/developer-broker-stream-connector.md:314]: Is it really true? If yes, this is a bug!

File: en/developer/developer-broker-stream-connector.md [hunk 2]
```diff
@@ -293,62 +293,66 @@ get hostnames, etc...
...
+    three keys, `index_id`, `host_id` and `service_id`.
+8.  `get_instance_name(instance_id)` that gets from the cache the
     instance name corresponding to the instance id.
-9.  ``get_metric_mapping(metric_id)`` that gets from the cache the
+9.  `get_metric_mapping(metric_id)` that gets from the cache the
     metric mapping object of the given metric id. The result is a table
-    containing two keys, ``metric_id`` and ``index_id``.
-10. ``get_service_description(host_id,service_id)`` that gets from the cache the
+    containing two keys, `metric_id` and `index_id`.
+10. `get_service_description(host_id,service_id)` (**API v1 only**) that gets from the cache the
```
Comment (bouda1) [inline en/developer/developer-broker-stream-connector.md:323]: I don't think think is true...

File: en/developer/developer-broker-stream-connector.md [hunk 3]
```diff
@@ -293,62 +293,66 @@ get hostnames, etc...
...
     cache, *nil* is returned.
-16. ``get_severity(host_id[,service_id])`` that gets the severity of a host or
+16. `get_severity(host_id[,service_id])` that gets the severity of a host or
     a service. If you only provide the *host_id*, we suppose you want to get
     a host severity. If a host or a service does not have any severity, the
     function returns a *nil* value.
+17. `get_host(host_id)` (**API v2 only**) that returns an associative array
+    proviing all the available information for one host from the cache.
+18. `get_service(host_id, service_id)` (**API v2 only**) that returns an associative array
+    proviing all the available information for one service from the cache.
```
Comment (bouda1) [inline en/developer/developer-broker-stream-connector.md:351]: In fact, we recommend these functions to be used with API v2, but they both work with API v1.

File: fr/developer/developer-broker-stream-connector.md [hunk 4]
```diff
@@ -293,62 +293,66 @@ get hostnames, etc...
...
+4.  `get_hostgroup_name(id)` that gets from the cache the host group name of
     the given id. This function returns a string or *nil* otherwise.
-5.  ``get_hostgroups(host_id)`` that gets the list of host groups containing the
+5.  `get_hostgroups(host_id)` that gets the list of host groups containing the
     host corresponding to *host_id*. The return value is an array of objects,
     each one containing two fields, *group_id* and *group_name*.
-6.  ``get_hostname(id)`` that gets from the cache the host name corresponding to
+6.  `get_hostname(id)` (**API v1 only**) that gets from the cache the host name corresponding to
     the given host id. This function returns a string with the host name or
     *nil* otherwise.
```
Comment (bouda1) [inline fr/developer/developer-broker-stream-connector.md:314]: Same remark as above.

File: fr/developer/developer-broker-stream-connector.md [hunk 5]
```diff
@@ -293,62 +293,66 @@ get hostnames, etc...
...
     instance name corresponding to the instance id.
-9.  ``get_metric_mapping(metric_id)`` that gets from the cache the
+9.  `get_metric_mapping(metric_id)` that gets from the cache the
     metric mapping object of the given metric id. The result is a table
-    containing two keys, ``metric_id`` and ``index_id``.
-10. ``get_service_description(host_id,service_id)`` that gets from the cache the
+    containing two keys, `metric_id` and `index_id`.
+10. `get_service_description(host_id,service_id)` (**API v1 only**) that gets from the cache the
     service description of the given pair host_id / service_id. This function
     returns a string or *nil* otherwise.
```
Comment (bouda1) [inline fr/developer/developer-broker-stream-connector.md:325]: Same remark as above.

File: fr/developer/developer-broker-stream-connector.md [hunk 6]
```diff
@@ -293,62 +293,66 @@ get hostnames, etc...
...
     cache, *nil* is returned.
-16. ``get_severity(host_id[,service_id])`` that gets the severity of a host or
+16. `get_severity(host_id[,service_id])` that gets the severity of a host or
     a service. If you only provide the *host_id*, we suppose you want to get
     a host severity. If a host or a service does not have any severity, the
     function returns a *nil* value.
+17. `get_host(host_id)` (**API v2 only**) that returns an associative array
+    proviing all the available information for one host from the cache.
+18. `get_service(host_id, service_id)` (**API v2 only**) that returns an associative array
+    proviing all the available information for one service from the cache.
```
Comment (bouda1) [inline fr/developer/developer-broker-stream-connector.md:351]: Same remark as above.

**Review by bouda1 — COMMENTED** [review]
Several remarks. But nothing really blocking.

**Pull request closed without merging**